<a href="https://colab.research.google.com/github/Duban1922/Dub_1922/blob/main/Copia_de_pandas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Library**

In [ ]:
import numpy as np
import pandas as pd



# Import dataset

In [ ]:
v = "/content/data11 (1).xlsx"

In [ ]:
n = pd.read_excel(v)

## EXPLORATION

In [ ]:
n.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 374 entries, 0 to 373
Data columns (total 10 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Cod                           374 non-null    int64  
 1   Área                          374 non-null    object 
 2   Casos delgadez                374 non-null    int64  
 3   Proporción delgadez           374 non-null    float64
 4   Casos exceso de peso          374 non-null    int64  
 5   Proporción exceso de peso     374 non-null    float64
 6   Casos Retraso en talla        374 non-null    int64  
 7   Proporción Retraso en talla   374 non-null    float64
 8   Total Poblacion               374 non-null    int64  
 9   Año                           374 non-null    int64  
dtypes: float64(3), int64(6), object(1)
memory usage: 29.3+ KB


In [ ]:
n.head()

Cod           Área  Casos delgadez  Proporción delgadez  \
0    1        Usaquén             102                  1.8   
1    2      Chapinero               9                  1.7   
2    3       Santa Fe              57                  2.5   
3    4  San Cristóbal             103                  2.7   
4    5           Usme             248                  2.2   

   Casos exceso de peso  Proporción exceso de peso  Casos Retraso en talla   \
0                  1075                       19.4                      626   
1                    96                       17.8                       64   
2                   415                       18.0                      340   
3                   774                       20.3                      579   
4                  2092                       18.2                     2019   

   Proporción Retraso en talla   Total Poblacion   Año  
0                          11.3             5528  2007  
1                          11.9              539  2007  
2                          14.7             2310  2007  
3                          15.2             3817  2007  
4                          17.6            11490  2007

In [ ]:
n.tail()

Cod                Área  Casos delgadez  Proporción delgadez  \
369   18  Rafael Uribe Uribe             290                  2.8   
370   19      Ciudad Bolívar             780                  2.8   
371   20             Sumapaz               5                  2.3   
372   21            Sin dato             509                  2.7   
373    0            Distrito            6949                  2.7   

     Casos exceso de peso  Proporción exceso de peso  Casos Retraso en talla   \
369                  2400                       23.0                      575   
370                  5971                       21.5                     1791   
371                    65                       30.2                       20   
372                  3996                       21.4                     1155   
373                 55835                       22.0                    14527   

     Proporción Retraso en talla   Total Poblacion   Año  
369                           5.5            10433  2023  
370                           6.5            27737  2023  
371                           9.3              215  2023  
372                           6.2            18699  2023  
373                           5.7           253500  2023

## CLEANING

In [ ]:
n.rename(columns= {"Año":"año"}, inplace = True)
n.rename(columns= {"cod":"cod"}, inplace = True)
n.rename(columns= {"Área":"area"}, inplace = True)
n.rename(columns= {"Casos delgadez":"cd"}, inplace = True)
n.rename(columns= {"Proporción delgadez":"pd"}, inplace = True)
n.rename(columns= {"Casos exceso de peso":"cep"}, inplace = True)
n.rename(columns= {"Proporción exceso de peso":"pep"}, inplace = True)
n.rename(columns= {"Casos Retraso en talla ":"crt"}, inplace = True)
n.rename(columns= {"Proporción Retraso en talla ":"prt"}, inplace = True)
n.rename(columns= {"Total Poblacion":"tp"}, inplace = True)




In [ ]:
n2 = n

In [ ]:
n.head()

Cod           area   cd   pd   cep   pep   crt   prt     tp   año
0    1        Usaquén  102  1.8  1075  19.4   626  11.3   5528  2007
1    2      Chapinero    9  1.7    96  17.8    64  11.9    539  2007
2    3       Santa Fe   57  2.5   415  18.0   340  14.7   2310  2007
3    4  San Cristóbal  103  2.7   774  20.3   579  15.2   3817  2007
4    5           Usme  248  2.2  2092  18.2  2019  17.6  11490  2007

##DELETE COLUMNS

In [ ]:
n2 = n2.drop(columns= ["Cod"])

In [ ]:
n2.head()

area   cd   pd   cep   pep   crt   prt     tp   año
0        Usaquén  102  1.8  1075  19.4   626  11.3   5528  2007
1      Chapinero    9  1.7    96  17.8    64  11.9    539  2007
2       Santa Fe   57  2.5   415  18.0   340  14.7   2310  2007
3  San Cristóbal  103  2.7   774  20.3   579  15.2   3817  2007
4           Usme  248  2.2  2092  18.2  2019  17.6  11490  2007

In [ ]:
n2["año_str"] = n2["año"].astype(str)

# **5: Indexing**



In [ ]:
n2 = n2.set_index(["area", "año"])

n2.head()

cd   pd   cep   pep   crt   prt     tp año_str
area          año                                                  
Usaquén       2007  102  1.8  1075  19.4   626  11.3   5528    2007
Chapinero     2007    9  1.7    96  17.8    64  11.9    539    2007
Santa Fe      2007   57  2.5   415  18.0   340  14.7   2310    2007
San Cristóbal 2007  103  2.7   774  20.3   579  15.2   3817    2007
Usme          2007  248  2.2  2092  18.2  2019  17.6  11490    2007

# **6. Descriptive stattistics**



In [ ]:
# Basic
n2[["cd", "cep", "crt"]].describe()

cd           cep           crt
count   374.00000    374.000000    374.000000
mean    299.57754   3073.727273    955.978610
std     831.55267   8086.462693   2564.773506
min       0.00000      0.000000      0.000000
25%      40.00000    396.000000    133.250000
50%     102.00000   1110.500000    339.500000
75%     213.50000   2258.000000    683.500000
max    9130.00000  83201.000000  26407.000000

In [ ]:
# Basic, but better format
n2[["cd", "cep", "crt"]].describe().applymap(lambda x: "{:,.1f}".format(x))

cd       cep       crt
count    374.0     374.0     374.0
mean     299.6   3,073.7     956.0
std      831.6   8,086.5   2,564.8
min        0.0       0.0       0.0
25%       40.0     396.0     133.2
50%      102.0   1,110.5     339.5
75%      213.5   2,258.0     683.5
max    9,130.0  83,201.0  26,407.0

## **Main Ideas of Those Descriptive Statistics**

Reviewing the description of the statistics, it is violated that in the localities of Bogotá there are more cases of excess weight than cases of thinness and cases of height retardation since it can be evidenced that in cases of excess weight they are more present in the percentiles and in the average; For this reason, it will be analyzed in detail because these types of cases (cep) are more present in the localities of Bogota.



##7.1 fILTERS


In [ ]:
# localidad
Usaquén = n2.filter( like = "Usaquén", axis = 0 )
Usaquén.head()

cd   pd   cep   pep  crt   prt     tp año_str
area    año                                                 
Usaquén 2007  102  1.8  1075  19.4  626  11.3   5528    2007
        2008  119  1.4  1708  20.4  730   8.7   8384    2008
        2009  177  2.1  1577  18.7  669   7.9   8428    2009
        2010  188  1.7  2322  20.9  882   7.9  11132    2010
        2011  167  1.6  2252  21.8  728   7.1  10318    2011

In [ ]:
## Filtrar una variable exacta
n2.query("area == 'Usme'")

cd   pd   cep   pep   crt   prt     tp año_str
area año                                                  
Usme 2007  248  2.2  2092  18.2  2019  17.6  11490    2007
     2008  116  2.3   778  15.5   734  14.6   5015    2008
     2009  169  2.1  1380  17.4  1092  13.7   7949    2009
     2010  203  2.5  1731  21.0  1251  15.2   8236    2010
     2011  211  2.7  1628  20.8  1108  14.1   7841    2011
     2012   91  1.7  1029  19.8   522  10.0   5203    2012
     2013  138  2.7  1012  19.9   447   8.8   5090    2013
     2014  112  2.1  1251  23.9   469   9.0   5234    2014
     2015  185  2.9  1447  22.3   449   6.9   6481    2015
     2016  103  1.4  1901  26.7   639   9.0   7132    2016
     2017   70  1.5  1325  27.9   417   8.8   4753    2017
     2018  187  2.8  1784  27.0   864  13.1   6596    2018
     2019  143  1.7  1973  24.1   547   6.7   8196    2019
     2020  239  2.5  2637  27.6  1064  11.1   9547    2020
     2021  335  2.5  3537  26.7  1130   8.5  13252    2021
     2022  469  2.6  4149  23.1  1292   7.2  17965    2022
     2023  365  2.9  2524  19.9   926   7.3  12714    2023

In [ ]:
# Solo traer un año
p2021 = n2.query("año == 2021")
p2021.head()

cd   pd   cep   pep   crt  prt     tp año_str
area          año                                                 
Usaquén       2021  224  2.2  2984  29.4   492  4.9  10142    2021
Chapinero     2021   86  2.2  1166  29.2   194  4.9   3992    2021
Santa Fe      2021  155  2.3  2022  29.7   384  5.6   6801    2021
San Cristóbal 2021  284  2.3  3667  30.1   759  6.2  12182    2021
Usme          2021  335  2.5  3537  26.7  1130  8.5  13252    2021

In [ ]:
n2.query("cd > 0")

cd   pd    cep   pep    crt   prt      tp año_str
area               año                                                      
Usaquén            2007   102  1.8   1075  19.4    626  11.3    5528    2007
Chapinero          2007     9  1.7     96  17.8     64  11.9     539    2007
Santa Fe           2007    57  2.5    415  18.0    340  14.7    2310    2007
San Cristóbal      2007   103  2.7    774  20.3    579  15.2    3817    2007
Usme               2007   248  2.2   2092  18.2   2019  17.6   11490    2007
...                       ...  ...    ...   ...    ...   ...     ...     ...
Rafael Uribe Uribe 2023   290  2.8   2400  23.0    575   5.5   10433    2023
Ciudad Bolívar     2023   780  2.8   5971  21.5   1791   6.5   27737    2023
Sumapaz            2023     5  2.3     65  30.2     20   9.3     215    2023
Sin dato           2023   509  2.7   3996  21.4   1155   6.2   18699    2023
Distrito           2023  6949  2.7  55835  22.0  14527   5.7  253500    2023

[370 rows x 8 columns]

## 7.2 Pivot


In [ ]:
## Be careful you must use the indexed dataframe (n)

## Example
### columnas = year
### columnas = codigo del pais
### = Varibale = Casos de peso

pivot1 = n.pivot(index = "año", columns = "Cod", values = "cep")
pivot1.head()

Cod      0     1    2    3     4     5    6     7     8     9   ...   12   13  \
año                                                             ...             
2007  39954  1075   96  415   774  2092  682  2809   266   216  ...  163  119   
2008  18885  1708  426  123   544   778  638  1505  2718   630  ...  709  335   
2009  21438  1577  194  334  1128  1380  679  1535  3037   869  ...  434  311   
2010  25971  2322  396  401  1665  1731  795  2196  3993  1058  ...  509  178   
2011  26512  2252  255  340  1458  1628  790  2334  3650  1106  ...  162  194   

Cod    14   15   16   17    18    19   20     21  
año                                               
2007  211  303  114  132  1148   187   43  26474  
2008  439  348  538  332   686   612   87      0  
2009   20  168  364   50   198   792  138   4003  
2010  354  411  480  155  1611  1161  188   2039  
2011  414  226  951  175  1410  1729   87   2126  

[5 rows x 22 columns]

## 8. GRAPHS

In [ ]:
import plotly.express as px

## 8.1 Line graph

In [ ]:
Usme = n2.query("area == 'Usme'")
Usme.head()

cd   pd   cep   pep   crt   prt     tp año_str
area año                                                  
Usme 2007  248  2.2  2092  18.2  2019  17.6  11490    2007
     2008  116  2.3   778  15.5   734  14.6   5015    2008
     2009  169  2.1  1380  17.4  1092  13.7   7949    2009
     2010  203  2.5  1731  21.0  1251  15.2   8236    2010
     2011  211  2.7  1628  20.8  1108  14.1   7841    2011

In [ ]:
# Fig 1
fig1 = px.line(Usme, x = 'año_str', y = 'cep', title = "Cases of excess weight 2007-2023")
fig1.show()

In [ ]:
# Fig 2
fig2 = px.line(Usme, x = 'año_str', y = 'cep')

#title
fig2.update_layout(title_text = "Cases of excess weight 2007-2023", title_font_size = 18, title_x = 0.5)


fig2.show()

#axis
fig2.update_xaxes(title = "year", title_font_size = 18)
fig2.update_yaxes(title = "Cases of excess weight 2007-2023", title_font_size = 18)

fig2.show()

In [ ]:
# Fig 3
fig3 = px.bar(Usme, x = 'año_str', y = 'cep')

# title

fig3.update_layout(title_text = 'Cases of excess weight in Usme',
                   title_font_size = 30,
                   title_x = 0.5)

# axis
fig3.update_xaxes(title = 'Years', title_font_size = 20)
fig3.update_yaxes(title = 'number of cases', title_font_size = 20)

fig3.show()

## **INTERPRETATION FIG 3**

The graph shows the number of cases of excess weight in the town of Usme from 2007 to 2023. The y-axis represents the number of cases, while the x-axis represents the year. The line on the graph shows a general upward trend, which means that the number of cases of excess weight has increased steadily over the years. This trend is worrying, as excess weight is a risk factor for a number of serious health problems, including diabetes, heart disease and some types of cancer.It is important to note that the graph only shows the number of diagnosed cases of excess weight. There are likely many more undiagnosed cases. Additionally, the graph does not take into account other factors that may contribute to excess weight.However, the graph provides a worrying picture of the current trend of excess weight in Usme. It is important to take steps to prevent excess weight and promote a healthy lifestyle.

## 8.3 More than one line

In [ ]:
## Create a list of countries
c_list = ['Usaquén', 'Chapinero', 'Santa fe', 'San Cristóbal', 'Usme', 'Ciudad Bolívar']

## Extract the countries from the dataframe
### First we must unindexed the df
n3 = n2.reset_index()
area = n3[n3['area'].isin(c_list)]

area.head()

area   año   cd   pd   cep   pep   crt   prt     tp año_str
0          Usaquén  2007  102  1.8  1075  19.4   626  11.3   5528    2007
1        Chapinero  2007    9  1.7    96  17.8    64  11.9    539    2007
3    San Cristóbal  2007  103  2.7   774  20.3   579  15.2   3817    2007
4             Usme  2007  248  2.2  2092  18.2  2019  17.6  11490    2007
18  Ciudad Bolívar  2007   22  2.1   187  18.0   144  13.9   1037    2007

In [ ]:
fig4 = px.line(area, x = 'año_str', y = 'crt', color = 'area')

# title

fig4.update_layout(title_text = 'Cases of delay in size in localities',
                   title_font_size = 30,
                   title_x = 0.5)

# axis
fig4.update_xaxes(title = 'Years', title_font_size = 20)
fig4.update_yaxes(title = 'Number of cases', title_font_size = 20)

fig4.show()

## **INTERPRETATION FIG 4**


The graph shows that the number of cases of stunting in locations in Bogota has increased significantly in the last 16 years. This trend is worrying, since stunting is a serious health problem that can have long-term consequences for a child's physical and cognitive development.The graph also shows that the number of cases of delayed height in localities varies depending on the locality. For example, in 2023, there were approximately 926 cases in Usaquén, while there were approximately 1,791 cases in Ciudad Bolívar. This variation may be due to a number of factors, such as socioeconomic status, access to healthcare, and diet quality.

It is important that health authorities investigate the causes of the variation in the number of stunting cases in localities and take measures to ensure that all children have access to the care and resources they need to grow and develop healthily.

## 8.4 Pie Chart


In [ ]:
# The idea is to graph the top 20 countries
# by population and the year 2020

p2021_2 = p2021.sort_values('cd', ascending = False)
p2021_2.head()

## Top 20 by population
p2021_2 = p2021_2.iloc[0:20]

In [ ]:
fig5 = px.pie(p2021_2, values = 'cd', names = p2021_2.index.get_level_values(0))

# title
fig5.update_layout(title_text = 'cases of thinness in 20 locations in Bogota',
                   title_font_size = 30,
                   title_x = 0.5)
fig5.show()

## **INTERPRETATION FIG 5**

The graph shows the distribution of cases of thinness in the 20 locations of Bogotá, Colombia. The graph is presented in the form of a pie chart, where each sector represents a locality and the size of the sector is proportional to the number of cases of thinness recorded in that locality.
The town with the highest number of cases of thinness is Kennedy, with 15.8% of the total.
They are followed by Suba, with 14.7%, and Engativá, with 1.33%.
The towns with the lowest number of cases of thinness are Bosa, with 1.46%, los Martires 1.33% and Antoni Nariño 0.94%; It also shows that the distribution of cases of thinness in the 20 localities of Bogotá is unequal. Some towns, such as Kennedy, Suba and Engativá, have a significantly higher concentration of cases of thinness than others.

This information can be useful for health authorities to identify localities that require greater attention and to develop prevention and care strategies for thinness.
